torch/_decomp/decompositions.py

```diff
@@ -1991,12 +1991,13 @@ def upsample_compute_output_size(input_size, output_size, scale_factors):
             lambda: "Must specify exactly one of output_size and scale_factors",
         )
         utils.check(len(scale_factors) == spatial_dimensions, lambda: "")
-        return [
-            # Returning output_size as float. We cannot convert it to int directly,
-            # as latter computation of scale_factor is relying output size being float
-            sym_float(input_size[i + 2] * scale_factors[i])
-            for i in range(spatial_dimensions)
-        ]
+        output_size = []
+        for i, s in enumerate(scale_factors):
+            if int(s) == s:
+                output_size.append(input_size[i + 2] * int(scale_factors[i]))
+            else:
+                output_size.append(sym_int(input_size[i + 2] * scale_factors[i]))
```

Comment:
is it safer to explicitly convert input_size into a double here? I guess Python automatically does the promotion